### PR TITLE
feat/restore_instant_listen

### PR DESCRIPTION
**Pull request by JarbasAl**
feat/restore_instant_listen config flag to not wait for listen sound to finish playing

### CODE REVIEW

**Comment by emphasize** [comment]
i do think we are doing additional work that should be handled earlier. Why it has [a sound] with `instant_listen == True`? check [here](https://github.com/OpenVoiceOS/ovos-dinkum-listener/blob/e798c0dea7142b8c6cef82def29376561b55175f/ovos_dinkum_listener/voice_loop/hotwords.py#L150) resp. assignment above

**Comment by emphasize** [comment]
The listen event implementation directly poses the question about `confirm_listening`/`instant_listen` config duplication

File: ovos_dinkum_listener/service.py [hunk 1]
```diff
@@ -727,9 +728,12 @@ def _handle_listen(self, message: Message):
...
                 message = message or Message("", context=context)  # might be None
                 self.bus.emit(message.forward("mycroft.audio.play_sound", {"uri": sound}))
-                self.voice_loop.state = ListeningState.CONFIRMATION
-                self.voice_loop.confirmation_event.clear()
-                Timer(0.5, lambda: self.voice_loop.confirmation_event.set()).start()
+                if not self.config["listener"].get("instant_listen"):
+                    self.voice_loop.state = ListeningState.CONFIRMATION
+                    self.voice_loop.confirmation_event.clear()
+                    Timer(0.5, lambda: self.voice_loop.confirmation_event.set()).start()
+                else:
```
Comment (emphasize) [inline ovos_dinkum_listener/service.py:735]: proposed change:

```python
if self.config.get('confirm_listening') and not self.config["listener"].get("instant_listen"):
        sound = self.config.get('sounds', {}).get('start_listening')
        if sound:
                ...
 ```

**Comment by JarbasAl** [comment]
> i do think we are doing additional work that should be handled earlier. Why does it have [a sound](https://github.com/OpenVoiceOS/ovos-dinkum-listener/blob/e798c0dea7142b8c6cef82def29376561b55175f/ovos_dinkum_listener/service.py#L566) attached with `instant_listen == True`? check [here](https://github.com/OpenVoiceOS/ovos-dinkum-listener/blob/e798c0dea7142b8c6cef82def29376561b55175f/ovos_dinkum_listener/voice_loop/hotwords.py#L150) resp. assignment above (this is the context that gets passed)

instant listen is still supposed to play the sound, just skip the waiting and assuming STT/VAD will handle it

**Review by emphasize — APPROVED** [review]

**Comment by goldyfruit** [comment]
Tested and it works, but the confirmation sound came almost at the end of my sentence.

**Review by j1nx — APPROVED** [review]
Can't test this myself, however @goldyfruit did succesfully so let's merge.

**Comment by JarbasAl** [comment]
> Can't test this myself, however @goldyfruit did succesfully so let's merge.

unittests failing, need to be updated

**Pull request merged**